bid-distribution/src/settlement_claims.rs

```diff
@@ -0,0 +1,112 @@
+
+    SettlementCollection {
+        slot: stake_meta_index.stake_meta_collection.slot,
+        epoch: stake_meta_index.stake_meta_collection.epoch,
+        settlements,
+    }
+}
+
+pub fn generate_bid_settlements(
+    stake_meta_index: &StakeMetaIndex,
+    sam_validator_metas: &Vec<ValidatorSamMeta>,
+    stake_authority_filter: &dyn Fn(&Pubkey) -> bool,
+    settlement_config: &SettlementConfig,
+) -> Vec<Settlement> {
+    info!("Generating bid settlements...");
+    let mut settlement_claim_collections = vec![];
+
+    for validator in sam_validator_metas {
+        if let Some(grouped_stake_metas) =
+            stake_meta_index.iter_grouped_stake_metas(&validator.vote_account)
+        {
+            if validator.effective_bid == 0.0 {
+                continue;
+            }
+            let sam_target_stake = sol_to_lamports(validator.marinade_sam_target_sol);
+            let mnde_target_stake = sol_to_lamports(validator.marinade_mnde_target_sol);
+            let max_wanted_stake = sol_to_lamports(validator.max_stake_wanted);
+            let marinade_payment_percentage = *settlement_config.marinade_fee_bps() as f64 / 10000.0;
+            
+            let marinade_stake: u64 = stake_meta_index.iter_grouped_stake_metas(&validator.vote_account).unwrap()
+                .filter(|(&(_, &stake_authority), _)| stake_authority_filter(&stake_authority))
+                .flat_map(|(_, metas)| metas.iter())
+                .map(|meta| meta.active_delegation_lamports)
+                .sum();
+
+            if sam_target_stake + mnde_target_stake == 0 {
+                continue;
+            }
+            let stake_sam_percentage = sam_target_stake as f64 / (sam_target_stake as f64 + mnde_target_stake as f64);
+            let initial_sam_stake =  (marinade_stake as f64 * stake_sam_percentage) as u64;
+            let effective_sam_stake: u64 = min(initial_sam_stake, max_wanted_stake);
+            let effective_total_bid_claim = (effective_sam_stake as f64 * validator.effective_bid) as u64;
+            let marinade_fee_claim = (effective_total_bid_claim as f64 * marinade_payment_percentage) as u64;
+            let stakers_total_claim = (effective_total_bid_claim as f64 * (1.0 - marinade_payment_percentage)) as u64;
+
+            let mut claims = vec![];
+            let mut claims_amount = 0;
+
+            for ((withdraw_authority, stake_authority), stake_metas) in grouped_stake_metas {
+                if !stake_authority_filter(stake_authority) {
+                    continue;
+                }
+                let stake_accounts: HashMap<_, _> = stake_metas
+                    .iter()
+                    .map(|s| (s.pubkey, s.active_delegation_lamports))
+                    .collect();
+                let active_stake: u64 = stake_accounts.values().sum();
+                let claim_amount = ((active_stake as f64 / effective_sam_stake as f64) * stakers_total_claim as f64) as u64;
```

Comment:
This seems to me wrong but correct me when otherwise.

The `active_stake` will be equal but rather bigger than the `effective_sam_stake`. From that the `claim_amount` will be bigger than `stakers_total_claim`, won't be?

Reply:
you're right, the ratio is reversed